src/Symfony/Component/Console/Helper/Table.php

```diff
@@ -205,24 +205,26 @@ public function setRow($column, array $row)
     public function render()
     {
         $this->calculateNumberOfColumns();
-        $this->rows = $this->buildTableRows($this->rows);
-        $this->headers = $this->buildTableRows($this->headers);
+        $rows = $this->buildTableRows($this->rows);
+        $headers = $this->buildTableRows($this->headers);
+
+        $this->calculateColumnsWidth(array_merge($headers, $rows));
```

Comment:
I'm not sure if it could change the state of the object as we call cleanup on each call https://github.com/aitboudad/symfony/blob/colspan_render_table/src/Symfony/Component/Console/Helper/Table.php#L571
we need to pass $columnWidths and $numberOfColumns to many methods ? does it make sense ?